### ISSUE TITLE
Pressing DoIt button without clicking on sourcecode area nor selecting anything does this ugly error

### ISSUE THREAD

**Issue by TazeTSchnitzel** [amber-smalltalk/trysmalltalk#25]
`Parse error on line 1 column 47 : Unexpected character null`

Not exactly beginner-friendly.

**Comment by hhzl** [comment]
Similar nasty things happen when pressing the ``PrintIt`` and ``DoIt`` buttons.

**Comment by tomrake** [comment]
This nasty bug is also present in helios.
## Reproduction
1. Get a new version of helios
2. Open a workspace or any code mirror pane.
3. Paste the following code into that space.
```
"A comment area one
two
three"
1+1 
```
4. Place the cursor at the very beginning of the code mirror pane ( that is before the first `"`.)
5. Click a "Print It" button.
6. You should get an Error: Unexpected character null

## My Guess on what is happening

1. The logic :An empty selection is not actionable therefore the user must mean act on the entire line.
2. The line is the beginning of a comment and is not closed
3. The parser throws an error.

## How to fix?

Handle errors from the parser in a better manner.

**Comment by herby** [comment]
@tomrake Yes, this and similar ones are parser errors. They are handled very rudimetarily by just spitting them out to user. If you think this is a problem in itself, there should probably be issues for this in the right repos (helios and amber-attic).

**Comment by hhzl** [comment]
In the case of trysmalltalk (not Helios workspace) the error occurs after the page has been loaded and no cursor is in the code pane. As soon as you clicked in the code pane and thus a cursor there the error message are more meaningful.

**Comment by hhzl** [comment]
Interestingly in case I have a cursor in a line and click ``doIt`` the whole line is executed. No error message.

**Comment by herby** [comment]
That's not "interestingly". That's how it works (no selection - run the actual line).
That's why I though that this thing can be fixed in trysmalltalk itself, by just putting the cursor into the area programmatically upon startup.

**Comment by hhzl** [comment]
## Bug analysis

We have two cases

a) 'no selection', i.e. no cursor in the source text area --- this is
the current default. If a button is pressed something (what actually?)
is evaluated which leads to a compiler error.

b) a cursor in the source text area. If there is no text selected but
just the cursor there then the whole line is evaluated.

## Proposed fix

Use b) as the default.

This is
"Put the cursor at a proper place after the page has loaded."

## Question

How do I put the cursor into a source text field after the page has loaded?
